src/hotspot/share/runtime/abstract_vm_version.cpp

```diff
@@ -128,11 +129,19 @@ const char* Abstract_VM_Version::vm_name() {
 #endif
 
 static const char vm_vendor_string[sizeof(VENDOR) < VENDOR_PADDING ? VENDOR_PADDING : sizeof(VENDOR)] = VENDOR;
+const char* Abstract_VM_Version::_vendor_branding_override = nullptr;
```

Comment:
Can't we just have:
```
const char* Abstract_VM_Version::_vm_vendor = vm_vendor-string; // Default unless overriden from VersionProps
const char* Abstract_VM_Version::vm_vendor() { return _vm_vendor; }
void Abstract_VM_Version::set_vm_vendor(const char* vm_vendor) {
  _vm_vendor = ...
}
```
???

Reply:
Or are you assuming/expecting the VersionProps value is null unless being overridden? You could initialize the VersionProps value to be the same as what VENDOR is at build time and always use it to set the Abstract_VM_Version field ... though I'm not sure if we really need it in Abstract_VM_Version ...

Reply:
never mind, the built value is still in vm_vendor_string. Since we should only call set_vm_vendor once, we don't worry about free.